include/mbedtls/platform.h

```diff
@@ -43,6 +43,8 @@
 #include "platform_time.h"
 #endif
 
+#define MBEDTLS_ERR_PLATFORM_HW_FAILED       -0x0080 /**< Hardware platform function failed. */
+
```

Comment:
Is this really the only error code we want to add?

How about "already in use/busy"? Or "Invalid operation"? I'd suggest out of memory, but I think we already have plans for that.

Reply:
I do not know what are the plans to handle the wide range of error codes that underlying platform/hardware might come across, but my guess is that each platform is most likely to end up returning their specific error codes. So I think its probably OK to just have this very generic error macro and add more as needed or see how we can integrate the platform errors with ours in some other way...

Reply:
I did have an "invalid operation" error code, but since I wasn't using it at the end, I didn't see a point of adding it.
We could think of many error codes to add, as the hw can fal on numerous reasons.  I  think that if we find ourselves in a need for other error codes, we could add. We already are low in available error values

Reply:
Yes, let's add the error codes when we have a need for them (not in this PR).